test/core.js

```diff
@@ -67,6 +67,48 @@ describe('snabbdom', function() {
       var vnode = h('a', {}, 'I am a string');
       assert.equal(vnode.text, 'I am a string');
     });
+    it('can create vnode with null props', function() {
+      var vnode = h('a', null);
+      assert.deepStrictEqual(vnode.data, {});
+      vnode = h('a', null, ['I am a string']);
+      assert.equal(vnode.children[0].text, 'I am a string');
+    });
+    it('can create vnode with children in vargs ', function() {
+      var vnode = h('a', null,
+        'some content',
+        h('span'),
+        ['part 1', 'part 2'],
+      );
+      assert.equal(JSON.stringify(vnode), JSON.stringify({
+        sel: 'a',
+        data: {},
+        children: [
+          {text: 'some content'},
+          {sel: 'span', data: {}},
+          {text: 'part 1'},
+          {text: 'part 2'},
+        ],
+      }));
+    });
+    it('can create vnode with falsey children removed', function() {
+      var vnode = h('a', null,
+        false,
+        [null, undefined],
+        'hello',
+        true,
+        [0, 1],
+      );
```

Comment:
Removal of falsey children is not an API I particularly like. Because it relaxes type checking in a way that instead of getting a proper type error when you've made a mistake, you get a missing element where you expect one.

In what case does JSX pass falsy values as children, please?

Reply:
The reason for removing falsey children is because this is a common usage

```ts
function render(showLogin: boolean, showCaptcha: boolean) {
  <div>
    {showLogin && <login-form />}
    {showCaptcha && <captcha-form />}
  </div>
}
```

which transpiles to

```
function render(showLogin, showCaptcha) {
    h("div", null,
        showLogin && h("login-form", null),
        showCaptcha && h("captcha-modal", null));
}
```

suppose showLogin and showCapcha are false, the call becomes `h('div', null, false, false)`. After flattening it becomes `h('div', null, [false, false])`. The expectation is the && if guards don't render `login-form` and `captcha-modal`. That's why I added boolean as primitive and was filtering it out.

My assumption is that it makes sense for the base h case as well. Generally most hyperscript/jsx based frameworks do filter out falsey (null, undefined, false) values due to the `xx && h(...)` kind of usage. 0 is a special case we don't want to filter because showing 0 in charts is common. Undefined will also be more common when `??` and `?.` (optional chaining) operators make it into spec. It's already stage3.

If you are against handling false, that is cool. We can only filter out null/undefined. That works with me.

in which case the usage will be

```
  <div>
    {showLogin ? <login-form /> : null}
    {showCaptcha  ? <captcha-form /> : null}
  </div>
```